### PR TITLE
Relations

### PR DESCRIPTION
**Pull request by Itswali**
### IN THIS PR.

- Create vets Table.
- Create specialization table
- Create visits Table.
- Relate all the Tables.
- Insert data in tables.

- [ ] IMAGE HERE.
`IMAGE - 1`

`IMAGE - 2`

`IMAGE - 3`

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -39,4 +39,27 @@ ALTER TABLE animals ADD owner_id INT;
...
 
 -- make owners_id foreign key with refrence from owners tbl
-ALTER TABLE animals ADD FOREIGN KEY (owner_id) REFERENCES owner(owner_id);
\ No newline at end of file
+ALTER TABLE animals ADD FOREIGN KEY (owner_id) REFERENCES owner(owner_id);
+
+-- FOR PART 4
+
+-- VETS Table Creation
+CREATE TABLE vets (id SERIAL PRIMARY KEY,
```
Comment (Alfredbis29) [inline schema.sql:47]: - [ ] Firstly, you have created a table for vets, which is a good start, but there are some differences in the columns you specified, I think You have missed the `date_of_graduation` column and used a different data type for `name.` Therefore, please make the following changes to your `vets` table: , I think this how your table should look like :wink:  :point_down: 
```

CREATE TABLE vets (
id int NOT NULL IDENTITY(1,1),
name varchar(255),
age int,
date_of_graduation date,
PRIMARY KEY (id)
);
```

File: schema.sql [hunk 2]
```diff
@@ -39,4 +39,27 @@ ALTER TABLE animals ADD owner_id INT;
...
+name VARCHAR(50) NOT NULL,
+age INT NOT NULL,
+date_of_registration date NOT NULL);
+
+-- Specialization Table Creation
+CREATE TABLE specialization (
```
Comment (Alfredbis29) [inline schema.sql]: - [ ] I can see that you did a great job on this project but you still need to improve on something for it to be completed, you have created a `specialization` table :wink: but the name of the table does not match the requirements :smile: Additionally, you have used a different data type for the `species_id` column, and the primary key should be a composite key made up of `species_id` and `vet_id.` Therefore, please make the following changes to your `specializations` table: for it to match all the requirements :wink:  
```

CREATE TABLE specializations (
species_id int,
vet_id int,
PRIMARY KEY (species_id, vet_id)
);
```

File: schema.sql [hunk 3]
```diff
@@ -39,4 +39,27 @@ ALTER TABLE animals ADD owner_id INT;
...
+species_id INT REFERENCES species(species_id),
+vet_id INT REFERENCES vets(id),
+PRIMARY KEY (species_id, vet_id)
+);
+
+-- Visits Table Creation.
+
+CREATE TABLE visits (
```
Comment (Alfredbis29) [inline schema.sql:61]: - [OPTIONAL ] Nice work so far :wink: You have created a `visits` table, but again the primary key should be a composite key made up of `animal_id` and 'visit_date.` Therefore, kindly make the following changes to your `visits` table: I think this is how it should be :wink:  :point_down: 
```

CREATE TABLE visits (
animal_id int,
vet_id int,
visit_date date,
PRIMARY KEY (animal_id, visit_date)
);
```

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
#  STATUS: REQUIRED CHANGES 
 ## Hi @Itswali ,
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

  
 ## Highlights 
All tables are created :white_check_mark: 
Created foreign keys based on the requirements :white_check_mark:
Created Primary keys for each table :white_check_mark:
Set up associations between models. :white_check_mark:
Good Job so far :+1:
 
 ## Required Changes 

- [x] _Check under the review  

 ## Optional suggestions 

- [x] Give a descriptive title to your PR , It will help us understand what you really work on for this milestone of your project :wink: , instead of saying `Relations` :wink: 
- [ ] Check under the review :+1: 

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
 
 Cheers and Happy coding! :clap: :clap: :clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Itswali) [reply schema.sql:47]: Thanks for noticing its resolved now.

**Comment by Itswali** [comment]
`Image of vets table.`

image

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Itswali 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**